2023-10-06 11:24:40 info: ---------------------------------------
2023-10-06 11:24:40 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:24:40 info: ---------------------------------------
2023-10-06 11:24:42 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable (1 ms)
    × should have all TODO comments removed (1 ms)
    √ `getAnonName` should not contain unneeded console.log calls (1 ms)
    √ should call `new Promise()`
    × should take a single argument (1 ms)
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should resolve when called with a string argument (1 ms)
    × should reject with an Error object when called without an argument (2 ms)

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● getAnonName › should take a single argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      58 |
      59 |   test('should take a single argument', () => {
    > 60 |     expect(state.paramCount).toBe(1);
         |                              ^
      61 |   });
      62 |
      63 |   test('`resolve()` should be called with a one argument', () => {

      at Object.toBe (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:60:30)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).rejects.toBeInstanceOf()

    Received promise resolved instead of rejected
    Resolved to value: "undefined Doe"

      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
    > 92 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |            ^
      93 |   });
      94 | });
      95 |

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:92:12)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 6 passed, 9 total
Snapshots:   0 total
Time:        0.373 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-10-06 11:24:43 error: *** ESLint Report ***

ex1-johnWho.js
  12:33  error    'callback' is defined but never used  no-unused-vars
  23:1   warning  Remove commented-out code             hyf/no-commented-out-code
  29:1   warning  Remove commented-out code             hyf/no-commented-out-code
  31:1   warning  Remove commented-out code             hyf/no-commented-out-code

✖ 4 problems (1 error, 3 warnings)


2023-10-06 11:25:24 info: ---------------------------------------
2023-10-06 11:25:24 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:25:24 info: ---------------------------------------
2023-10-06 11:25:25 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable
    √ should have all TODO comments removed
    √ `getAnonName` should not contain unneeded console.log calls (1 ms)
    √ should call `new Promise()`
    × should take a single argument (1 ms)
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should resolve when called with a string argument (1 ms)
    × should reject with an Error object when called without an argument

  ● getAnonName › should take a single argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      58 |
      59 |   test('should take a single argument', () => {
    > 60 |     expect(state.paramCount).toBe(1);
         |                              ^
      61 |   });
      62 |
      63 |   test('`resolve()` should be called with a one argument', () => {

      at Object.toBe (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:60:30)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).rejects.toBeInstanceOf()

    Received promise resolved instead of rejected
    Resolved to value: "undefined Doe"

      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
    > 92 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |            ^
      93 |   });
      94 | });
      95 |

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:92:12)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 7 passed, 9 total
Snapshots:   0 total
Time:        0.288 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-10-06 11:25:26 error: *** ESLint Report ***

ex1-johnWho.js
  12:33  error    'callback' is defined but never used  no-unused-vars
  23:1   warning  Remove commented-out code             hyf/no-commented-out-code
  29:1   warning  Remove commented-out code             hyf/no-commented-out-code
  31:1   warning  Remove commented-out code             hyf/no-commented-out-code

✖ 4 problems (1 error, 3 warnings)


2023-10-06 11:26:07 info: ---------------------------------------
2023-10-06 11:26:07 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:26:07 info: ---------------------------------------
2023-10-06 11:26:08 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable
    √ should have all TODO comments removed
    √ `getAnonName` should not contain unneeded console.log calls (1 ms)
    √ should call `new Promise()`
    √ should take a single argument
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should resolve when called with a string argument (1 ms)
    × should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).rejects.toBeInstanceOf()

    Received promise resolved instead of rejected
    Resolved to value: "undefined Doe"

      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
    > 92 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |            ^
      93 |   });
      94 | });
      95 |

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:92:12)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.287 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-10-06 11:26:09 error: *** ESLint Report ***

ex1-johnWho.js
  23:1  warning  Remove commented-out code  hyf/no-commented-out-code
  29:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2023-10-06 11:26:50 info: ---------------------------------------
2023-10-06 11:26:50 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:26:50 info: ---------------------------------------
2023-10-06 11:26:51 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable
    √ should have all TODO comments removed
    √ `getAnonName` should not contain unneeded console.log calls
    √ should call `new Promise()` (1 ms)
    √ should take a single argument
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should resolve when called with a string argument (1 ms)
    × should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).rejects.toBeInstanceOf()

    Received promise resolved instead of rejected
    Resolved to value: "undefined Doe"

      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
    > 92 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |            ^
      93 |   });
      94 | });
      95 |

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:92:12)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.241 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-10-06 11:26:52 error: *** ESLint Report ***

ex1-johnWho.js
  23:1  warning  Remove commented-out code  hyf/no-commented-out-code
  29:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2023-10-06 11:27:28 info: ---------------------------------------
2023-10-06 11:27:28 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:27:28 info: ---------------------------------------
2023-10-06 11:27:29 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable (1 ms)
    √ should have all TODO comments removed (1 ms)
    √ `getAnonName` should not contain unneeded console.log calls
    √ should call `new Promise()`
    √ should take a single argument
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should resolve when called with a string argument (1 ms)
    × should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).rejects.toBeInstanceOf()

    Received promise resolved instead of rejected
    Resolved to value: "undefined Doe"

      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
    > 92 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |            ^
      93 |   });
      94 | });
      95 |

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:92:12)

  ● getAnonName › should reject with an Error object when called without an argument

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      82 |
      83 |   test('should reject with an Error object when called without an argument', async () => {
    > 84 |     expect.assertions(3);
         |            ^
      85 |     expect(exported).toBeDefined();
      86 |     const timeoutSpy = jest
      87 |       .spyOn(global, 'setTimeout')

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:84:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.286 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-10-06 11:27:30 error: *** ESLint Report ***

ex1-johnWho.js
  22:1  warning  Remove commented-out code  hyf/no-commented-out-code
  28:1  warning  Remove commented-out code  hyf/no-commented-out-code
  30:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2023-10-06 11:30:19 info: ---------------------------------------
2023-10-06 11:30:19 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:30:19 info: ---------------------------------------
2023-10-06 11:30:20 info: All unit tests passed.
2023-10-06 11:30:21 error: *** ESLint Report ***

ex1-johnWho.js
  25:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code
  33:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2023-10-06 11:30:46 info: ---------------------------------------
2023-10-06 11:30:46 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:30:46 info: ---------------------------------------
2023-10-06 11:30:48 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable
    √ should have all TODO comments removed
    √ `getAnonName` should not contain unneeded console.log calls (1 ms)
    √ should call `new Promise()`
    √ should take a single argument
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should resolve when called with a string argument (1 ms)
    × should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should reject with an Error object when called without an argument

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "You didn't pass in a first name!"

      90 |     timeoutSpy.mockRestore();
      91 |     expect(promise).toBeInstanceOf(Promise);
    > 92 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      93 |   });
      94 | });
      95 |

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:241:22)
      at Object.toBeInstanceOf (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:92:36)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.294 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-10-06 11:30:49 error: *** ESLint Report ***

ex1-johnWho.js
  25:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code
  33:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2023-10-06 11:31:05 info: ---------------------------------------
2023-10-06 11:31:05 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 11:31:05 info: ---------------------------------------
2023-10-06 11:31:06 info: All unit tests passed.
2023-10-06 11:31:07 error: *** ESLint Report ***

ex1-johnWho.js
  25:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code
  33:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2023-10-06 12:06:07 info: ---------------------------------------
2023-10-06 12:06:07 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-10-06 12:06:07 info: ---------------------------------------
2023-10-06 12:06:08 info: All unit tests passed.
2023-10-06 12:06:10 info: All steps were completed successfully
2023-10-06 12:12:35 info: -------------------------------------------------
2023-10-06 12:12:35 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2023-10-06 12:12:35 info: -------------------------------------------------
2023-10-06 12:12:56 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js (20.323 s)
  checkDoubleDigits
    √ should exist and be executable
    √ should have all TODO comments removed
    √ `checkDoubleDigits` should not contain unneeded console.log calls
    √ should call new Promise() (1 ms)
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should be a function that takes a single argument
    × (9) should return a rejected promise with an Error object (5009 ms)
    × (10) should return a promise that resolves to "This is a double digit number!" (5009 ms)
    × (99) should return a promise that resolves to "This is a double digit number!" (5004 ms)
    × (100) should return a rejected promise with an Error object (5013 ms)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      61 |   });
      62 |
    > 63 |   test('(9) should return a rejected promise with an Error object', () => {
         |   ^
      64 |     expect.assertions(2);
      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);

      at test (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:63:3)
      at Object.describe (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:10:1)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      62 |
      63 |   test('(9) should return a rejected promise with an Error object', () => {
    > 64 |     expect.assertions(2);
         |            ^
      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);
      67 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:64:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      68 |   });
      69 |
    > 70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
         |   ^
      71 |     expect.assertions(3);
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);

      at test (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:70:3)
      at Object.describe (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:10:1)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      78 |   });
      79 |
    > 80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
         |   ^
      81 |     expect.assertions(2);
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);

      at test (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:80:3)
      at Object.describe (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:10:1)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      87 |   });
      88 |
    > 89 |   test('(100) should return a rejected promise with an Error object', () => {
         |   ^
      90 |     expect.assertions(2);
      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);

      at test (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:89:3)
      at Object.describe (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:10:1)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      88 |
      89 |   test('(100) should return a rejected promise with an Error object', () => {
    > 90 |     expect.assertions(2);
         |            ^
      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);
      93 |     return expect(promise).rejects.toBeInstanceOf(Error);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:90:12)

Test Suites: 1 failed, 1 total
Tests:       4 failed, 7 passed, 11 total
Snapshots:   0 total
Time:        20.364 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2023-10-06 12:13:12 info: -------------------------------------------------
2023-10-06 12:13:12 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2023-10-06 12:13:12 info: -------------------------------------------------
2023-10-06 12:13:13 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    √ should exist and be executable (1 ms)
    √ should have all TODO comments removed (1 ms)
    √ `checkDoubleDigits` should not contain unneeded console.log calls
    √ should call new Promise() (1 ms)
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument
    √ should be a function that takes a single argument
    √ (9) should return a rejected promise with an Error object (1 ms)
    × (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    × (99) should return a promise that resolves to "This is a double digit number!"
    √ (100) should return a rejected promise with an Error object

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 10]

      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);
    > 75 |     return expect(promise).resolves.toEqual(
         |            ^
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );
      78 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:75:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 99]

      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
    > 84 |     return expect(promise).resolves.toEqual(
         |            ^
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );
      87 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:84:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 9 passed, 11 total
Snapshots:   0 total
Time:        0.293 s, estimated 21 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2023-10-06 12:14:10 info: -------------------------------------------------
2023-10-06 12:14:10 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2023-10-06 12:14:10 info: -------------------------------------------------
2023-10-06 12:14:11 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    √ should exist and be executable (1 ms)
    √ should have all TODO comments removed
    √ `checkDoubleDigits` should not contain unneeded console.log calls
    √ should call new Promise()
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument (1 ms)
    √ should be a function that takes a single argument
    √ (9) should return a rejected promise with an Error object
    × (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    × (99) should return a promise that resolves to "This is a double digit number!"
    √ (100) should return a rejected promise with an Error object

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 10]

      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);
    > 75 |     return expect(promise).resolves.toEqual(
         |            ^
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );
      78 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:75:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 99]

      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
    > 84 |     return expect(promise).resolves.toEqual(
         |            ^
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );
      87 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:84:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 9 passed, 11 total
Snapshots:   0 total
Time:        0.247 s, estimated 1 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2023-10-06 12:14:43 info: -------------------------------------------------
2023-10-06 12:14:43 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2023-10-06 12:14:43 info: -------------------------------------------------
2023-10-06 12:14:44 info: All unit tests passed.
2023-10-06 12:14:46 info: All steps were completed successfully
